cmd/ctl/pkg/install/install.go

```diff
@@ -0,0 +1,187 @@
+Most of the features supported by 'helm install' are also supported by this command.
+Additional the the functionallity that the helm command gives you, this command will
+also manage CRD resources.
+
+Some example uses:
+	$ kubectl cert-manager install
+or
+	$ kubectl cert-manager install --version v1.4.0
+or
+	$ kubectl cert-manager install --set prometheus.enabled=false
+or
+	$ kubectl cert-manager install --namespace cert-manager-namespace
+
+To override values in the cert-manager chart, use either the '--values' flag and pass in a file
+or use the '--set' flag and pass configuration from the command line, to force
+a string value use '--set-string'. In case a value is large and therefore
+you want not to use neither '--values' nor '--set', use '--set-file' to read the
+single large value from file.
+`
+
+func NewCmdInstall(ctx context.Context, ioStreams genericclioptions.IOStreams, factory cmdutil.Factory) *cobra.Command {
+	settings := cli.New()
+	cfg := new(action.Configuration)
+	if err := cfg.Init(settings.RESTClientGetter(), settings.Namespace(), os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+		log.Printf("%+v", err)
+		os.Exit(1)
+	}
+
+	options := &InstallOptions{
+		settings:  settings,
+		client:    action.NewInstall(cfg),
+		cfg:       cfg,
+		valueOpts: &values.Options{},
+		IOStreams: ioStreams,
+	}
+
+	cmd := &cobra.Command{
+		Use:   "install",
+		Short: "install cert-manager",
+		Long:  installDesc,
+		RunE: func(_ *cobra.Command, args []string) error {
+			return options.runInstall(ctx)
+		},
+		SilenceUsage: true,
+	}
+
+	addInstallFlags(cmd.Flags(), options.client)
+	addValueOptionsFlags(cmd.Flags(), options.valueOpts)
+	addChartPathOptionsFlags(cmd.Flags(), &options.client.ChartPathOptions)
+
+	cmd.Flags().StringVar(&options.ChartName, "chart-name", "cert-manager", "name of the cert-manager chart")
+
+	return cmd
+}
+
+func (o *InstallOptions) runInstall(ctx context.Context) error {
+	log.SetFlags(0)
+	log.SetOutput(o.Out)
+
+	// 1. find chart
+	cp, err := o.client.ChartPathOptions.LocateChart(o.ChartName, o.settings)
+	if err != nil {
+		return err
+	}
+
+	chart, err := helm.LoadChart(cp, o.client, o.settings)
+	if err != nil {
+		return err
+	}
+
+	// Check if chart is installable
+	if err := helm.CheckIfInstallable(chart); err != nil {
+		return err
+	}
+
+	// Console print if chart is deprecated
+	if chart.Metadata.Deprecated {
+		log.Printf("This chart is deprecated")
+	}
+
+	// Merge all values flags
+	p := getter.All(o.settings)
+	chartValues, err := o.valueOpts.MergeValues(p)
+	if err != nil {
+		return err
+	}
+
+	// 2. do dryrun template generation (used for rendering the crds in /templates)
+	o.client.DryRun = true
+	o.client.IsUpgrade = true
+	chartValues[installCRDsFlagName] = true
+	dryRunResult, err := o.client.Run(chart, chartValues)
+	if err != nil {
+		return err
+	}
+	resouces, err := helm.GetChartResourceInfo(chart, dryRunResult.Manifest, true, o.cfg.KubeClient)
```

Comment:
I added some comments, the argument is set to true; because when installing we want to gather all crd-sources and from that do a crd-only install. Although, like explained in my comments, the parameter is only useful in case the /crds folder is used in the chart. So, it is possible to switch to such a system in case that would be a requirement.

Reply:
I'll maybe remove that part, because it is not used by cert-manager.